### PR TITLE
first implementation for economic metrics exporter sidecar

### LINKED ISSUE #82: Package exported economic metrics as self-contained sidecar binary
**Is your feature request related to a problem? Please describe.**

The node spec currently requires [eigen_registered_stakes](https://eigen.nethermind.io/docs/spec/metrics/metrics-prom-spec#economics-metrics) to be exported. We have made this available in the [economic collector](https://github.com/Layr-Labs/eigensdk-go/blob/master/metrics/collectors/economic/economic.go) which periodically calls the chain, but this limits its use. For eg, teams that don't use golang cannot use it.

**Describe the solution you'd like**

Exported metrics are self contained and should be packaged as a separate binary to be run as a sidecar. We should do this.

### PR DESCRIPTION
**Pull request by samlaf**
Fixes #82 .

### Solution
This works for now, but using the old ugly constructor.
Will switch to new constructor after we merge the chainio refactor PR #73 .

### CODE REVIEW

File: cmd/metrics_exporter/main.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(economicCollector)
+
+	metricsListeningAddr := c.String(MetricsListeningAddrFlag.Name)
+	logger.Infof("Starting metrics server at port %v", metricsListeningAddr)
+	http.Handle("/metrics", promhttp.HandlerFor(
+		reg,
+		promhttp.HandlerOpts{},
+	))
+	err = http.ListenAndServe(metricsListeningAddr, nil)
```
Comment (NimaVaziri) [inline cmd/metrics_exporter/main.go:145]: Thoughts on wrapping the http call around a `logger.Fatal` instead?

File: cmd/metrics_exporter/main.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		return err
+	}
+
+	// TODO: we hardcode a bunch of things for now, waiting for chainio refactor PR to be merged
+	//      https://github.com/Layr-Labs/eigensdk-go/pull/73
+	ethHttpUrl := c.String(RpcUrlFlag.Name)
+	// we don't even need this but are forced to pass one because constructor builds the websocket clients
+	ethWsUrl := "ws" + ethHttpUrl[4:]
+	constructorConfig := constructor.Config{
+		// we don't need writers for this exporter, but still need to pass this fake privatekey..
```
[the remark this answers is not in the document]
Comment (samlaf) [reply cmd/metrics_exporter/main.go:109]: I don't remember myself if it's there haha, but refactoring the constructor is definitely a todo. I still have this PR open as a suggestion: https://github.com/Layr-Labs/eigensdk-go/pull/11

File: cmd/metrics_exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Metrics Exporter Sidecar
```
[the remark this answers is not in the document]
Comment (samlaf) [reply cmd/metrics_exporter/README.md:1]: Yep. moved PR to other repo and added docker stuff there. see https://github.com/Layr-Labs/eigenlayer-middleware-offchain/pull/2

**Comment by samlaf** [comment]
@NimaVaziri I propose closing this PR, as @shrimalmadhur convinced me it doesn't belong in this repo.
Moved the PR over to https://github.com/Layr-Labs/eigenlayer-middleware-offchain/pull/2

**Comment by NimaVaziri** [comment]
Sgtm

**Pull request closed without merging**